locale/en/docs/guides/dont-block-the-event-loop.md

```diff
@@ -0,0 +1,435 @@
+I/O tasks involve querying an external service provider (database, file system, etc.) and waiting for its response.
+While an I/O-WP Worker is waiting for its response, it has nothing else to do and can be de-scheduled by the operating system, giving another Worker a chance to submit their request.
+*I/O tasks will make progress even while the associated thread is not running*.
+External service providers like databases and file systems have been highly optimized to handle many pending requests concurrently.
+For example, a file system will examine a large set of pending write and read requests to merge conflicting updates and to retrieve files in an optimal order (e.g. see [these slides](http://researcher.ibm.com/researcher/files/il-AVISHAY/01-block_io-v1.3.pdf)).
+
+If you have only one Worker Pool, e.g. the Node Worker Pool, then the differing characteristics of CPU-bound and I/O-bound work will harm your application's performance.
+
+#### Conclusions
+For simple tasks, like iterating over the elements of an arbitrarily long array, partitioning might be a good option.
+If your computation is more complex, offloading is a better approach: the communication costs, i.e. the overhead of passing serialized objects between the Event Loop and the Worker Pool, are offset by the benefit of using multiple cores.
+
+However, if your server relies heavily on complex calculations, you should think about whether Node is really a good fit. Node excels for I/O-bound work, but for expensive computation it might not be the best option.
+
+If you take the offloading approach, see the section on not blocking the Worker Pool.
+
+## Don't block the Worker Pool
+Node has a Worker Pool composed of `k` Workers.
+If you are using the Offloading paradigm discussed above, you might have your own Computational Worker Pool, to which the same principles apply.
+In either case, let us assume that `k` is much smaller than the number of clients you might be handling concurrently.
+This is in keeping with Node's "one thread for many clients" philosophy, the secret to its scalability.
+
+The Workers of this Pool use the cooperative multitasking approach discussed above: each Worker completes its current Task before proceeding to the next one on the Worker Pool queue.
+
+Now, there will be variation in the cost of the Tasks required to handle your clients' requests.
+Some Tasks can be completed quickly (e.g. reading short or cached files in the I/O-WP, or computing the average of a small array in the C-WP), and others will take longer (e.g reading larger or uncached files in the I/O-WP, or doing more expensive computation in the C-WP).
+Your goal should be to *minimize the variation in Task times*, and you should use *Task partitioning* to accomplish this.
+
+### What is the Node Worker Pool used for?
+In the Node core modules, the Node Worker Pool is used for the asynchronous versions of the following classes of APIs:
+1. File system I/O
+2. DNS queries
```

Comment:
only dns.lookup, the other ones don't use the pool

Reply:
Yes. I'll fix this. I believe all of the FS APIs map directly to libuv versions, though?

Reply:
https://nodejs.org/api/fs.html#fs_threadpool_usage  - except fs.FSWatcher()